crates/mongo-wasmtime/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+//! Implement the mongodb interface using the mongo-rust-driver (https://docs.rs/mongodb/latest/mongodb)
+//! This is using a Wasmtime host implementation.
+
+pub use wasi_mongo::add_to_linker;
+
+use wasi_mongo::*;
+use std::sync::Arc;
+
+use mongodb::sync::{Client,Database};
+
+wit_bindgen_wasmtime::export!("wit/ephemeral/wasi-mongo.wit");
+
+#[derive(Clone)]
+pub struct WasiMongoClient {
+    address: String,
+    client: Arc<Client>
+}
+
+impl wasi_mongo::WasiMongo for WasiMongoClient {
+    fn collection_exists(&mut self, db: &str, name: &str) -> Result<bool, Error>{
+        if let Ok(()) = self.collection_exists(db, name) {
+            return Ok(true)
+        } 
+        Err(Error::Error) 
+    }
+}
+
+impl WasiMongoClient {
+    pub fn new(addr: &str) -> anyhow::Result<Self> {
+        let client = Arc::new(Client::with_uri_str(addr)?);
+        Ok(WasiMongoClient {
+            address: addr.to_string(),
+            client
+        })
+    }
+
+    fn collection_exists(&mut self, db: &str, _name: &str) -> anyhow::Result<()> {
```

Comment:
nit: not the biggest fan of having the same name for multiple functions, distinguished by the trait they implement.
(not something to change right now)